bundles/org.openhab.binding.vizio/src/main/java/org/openhab/binding/vizio/internal/handler/VizioHandler.java

```diff
@@ -0,0 +1,572 @@
+        this.communicator = new VizioCommunicator(httpClient, EMPTY, -1, EMPTY);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Vizio handler");
+        final Gson gson = new Gson();
+        VizioConfiguration config = getConfigAs(VizioConfiguration.class);
+
+        @Nullable
+        String host = config.hostName;
+        final @Nullable String authToken = config.authToken;
+        @Nullable
+        String appListJson = config.appListJson;
+
+        if (host == null || host.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Host Name must be specified");
+            return;
+        } else if (host.contains(":")) {
+            // format for ipv6
+            host = "[" + host + "]";
+        }
+
+        this.communicator = new VizioCommunicator(httpClient, host, config.port, authToken != null ? authToken : EMPTY);
+
+        // register trustmanager service to allow httpClient to accept self signed cert from the Vizio TV
+        VizioTlsTrustManagerProvider tlsTrustManagerProvider = new VizioTlsTrustManagerProvider(
+                host + ":" + config.port);
+        serviceRegistration = FrameworkUtil.getBundle(getClass()).getBundleContext()
+                .registerService(TlsTrustManagerProvider.class.getName(), tlsTrustManagerProvider, null);
+
+        if (authToken == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING,
+                    "Auth Token must be specified, see README for instructions");
+            return;
+        }
+
+        // if app list is not supplied in thing configuration, populate it from the json db
+        if (appListJson == null) {
+            appListJson = dbAppsJson;
+
+            // Update thing configuration (persistent) - store app list from db into thing so the user can update it
+            Configuration configuration = this.getConfig();
+            configuration.put(PROPERTY_APP_LIST_JSON, appListJson);
+            this.updateConfiguration(configuration);
+        }
+
+        try {
+            VizioApps appsFromJson = gson.fromJson(appListJson, VizioApps.class);
+            if (appsFromJson != null && !appsFromJson.getApps().isEmpty()) {
+                userConfigApps = appsFromJson.getApps();
+
+                List<StateOption> appListOptions = new ArrayList<>();
+                userConfigApps.forEach(app -> {
+                    appListOptions.add(new StateOption(app.getName(), app.getName()));
+                });
+
+                stateDescriptionProvider.setStateOptions(new ChannelUID(getThing().getUID(), ACTIVE_APP),
+                        appListOptions);
+            }
+        } catch (JsonSyntaxException e) {
+            logger.debug("Invalid App List Configuration in thing configuration. Exception: {}", e.getMessage(), e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING,
```

Comment:
I do not think this detail is correct - please see:
https://www.openhab.org/docs/concepts/things.html#status-details

I assume the binding is not transferring configuration information to the device here?

Reply:
It would happen if the user modified the JSON and made it un-parsable. Maybe HANDLER_INITIALIZING_ERROR would be better?

Reply:
In that case **CONFIGURATION_ERROR** seems appropriate? **HANDLER_INITIALIZING_ERROR** is used by the framework with a corresponding status **UNINITIALIZED**.

Reply:
changed to CONFIGURATION_ERROR, thanks.